ERCS/erc-7754.md

```diff
@@ -0,0 +1,256 @@
+
+Currently, the communication channel between dapps and wallets is vulnerable to man in the middle attacks.
+Specifically, attackers can intercept RPC requests by injecting JavaScript code in the page,
+via e.g. an XSS vulnerability or due to a malicious extension.
+Once an RPC request is intercepted, it can be modified in a number of pernicious ways, including:
+
+- Editing the calldata in order to siphon funds or otherwise change the transaction outcome
+- Modifying the parameters of an [EIP-712](./eip-712.md) request
+- Obtaining a replayable signature from the wallet
+
+Even if the user realizes that requests from the dapp may be tampered with, they have little to no recourse to mitigate the problem.
+Overall, the lack of a chain of trust between the dapp and the wallet hurts the ecosystem as a whole:
+
+- Users cannot simply trust otherwise honest dapps, and are at risk of losing funds
+- Dapp maintainers are at risk of hurting their reputations if an attacker finds a viable MITM attack
+
+For these reasons, we recommend that wallets implement the `wallet_signedRequest` RPC method.
+This method provides dapp developers with a way to explicitly ask the wallet to verify the
+integrity of a payload. This is a significant improvement over the status quo, which forces
+dapps to rely on implicit approaches such as argument bit packing.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Overview
+
+We propose to use the dapp's domain certificate of a root of trust to establish a trust chain as follow:
+
+1. The user's browser verifies the domain certificate and displays appropriate warnings if overtaken
+2. The DNS record of the dapp hosts a TXT field pointing to a URL where a JSON manifest is hosted
+   - This file SHOULD be at a well known address such as `https://domain.com/.well-known/twit.json`
+3. The config file contains an array of objects of the form `{ id, alg, publicKey }`
+4. For signed requests, the dapp first securely signs the payload with a private key, for example by submitting a request to its backend
+5. The original payload, signature, and public key id are sent to the wallet via the `wallet_signedRequest` RPC method
+6. The wallet verifies the signature before processing the request normally
+
+### Wallet integration
+
+#### Key discovery
+
+Attested public keys are necessary for the chain of trust to be established.
+Since this is traditionally done via DNS certificates, we propose the addition of a DNS record containing the public keys.
+This is similar to DKIM, but the use of a manifest file provides more flexibility for future improvements, as well as support for multiple algorithm and key pairs.
+
+Similarly to standard JWT practices, the wallet could eagerly cache dapp keys.
```

Comment:
Similar comment here about JWT. Either expand the abbreviation or link to a definition.